### ISSUE TITLE
Request for 5x5 boxs

### ISSUE THREAD

**Issue by Days-gone** [TravisBubb/DotsAndBoxesAI#1]
hello , when it comes to 5x5 boxs , is there any way to make it fast enough to enjoy a  game ?

**Comment by TravisBubb** [comment]
To be honest with you, I am not sure if there is with this current implementation. I don't know how familiar you are with the minimax algorithm but it is not very efficient when it comes to bigger states.  There are simply too many paths to search through to be efficient, even with the added boost of alpha-beta pruning.

To put it simply, the fact that Python is being used and the nature of the algorithm - no. However, there certainly could be a much more efficient solution in a different AI implementation. Perhaps a more modern approach such as training a machine learning model.

Another thing that could help is studying various strategies that are good for earlier in the game. The earliest moves take the longest since there are more open spots. If you were to make a separate model with early game strategies that the agent could simply reference as needed (and not need to do any additional computations) then the AI would become a little more efficient, and also better at the game too.

**Comment by Days-gone** [comment]
Sincerely thank you for your helpful reply, I am trying imply&nbsp;Monte Carlo method&nbsp; to the early game ,and may try minmax algorithm in the endgame.​


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "TravisBubb/DotsAndBoxesAI"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年7月21日(星期四) 上午7:08
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [TravisBubb/DotsAndBoxesAI] Request for 5x5 boxs (Issue #1)





 
To be honest with you, I am not sure if there is with this current implementation. I don't know how familiar you are with the minimax algorithm but it is not very efficient when it comes to bigger states.  There are simply too many paths to search through to be efficient, even with the added boost of alpha-beta pruning.
 
To put it simply, the fact that Python is being used and the nature of the algorithm - no. However, there certainly could be a much more efficient solution in a different AI implementation. Perhaps a more modern approach such as training a machine learning model.
 
Another thing that could help is studying various strategies that are good for earlier in the game. The earliest moves take the longest since there are more open spots. If you were to make a separate model with early game strategies that the agent could simply reference as needed (and not need to do any additional computations) then the AI would become a little more efficient, and also better at the game too.

**Comment by TravisBubb** [comment]
Good luck!
